raft/raft.go

```diff
@@ -434,6 +448,7 @@ func (r *raft) reset(term uint64) {
 		}
 	}
 	r.pendingConf = false
+	r.readOnly = newReadOnly(r.readOnly.option)
```

Comment:
How would the application retry if you never tell it the node lost its leadership and the read failed? Is the application expected to watch SoftState.Status to see when leadership changes, or would it retry based on a timeout?